test/unittests/eof_stack_validation_test.cpp

```diff
@@ -12,145 +12,1214 @@
 using namespace evmone;
 using namespace evmone::test;
 
+namespace
+{
+// code prologue that creates a segment starting with possible stack heights 1 and 3
+const auto varstack = push0() + rjumpi(2, 0) + OP_PUSH0 + OP_PUSH0;
+}  // namespace
+
+TEST_F(eof_validation, unreachable_instructions)
+{
+    add_test_case(
+        eof_bytecode(bytecode{OP_STOP} + OP_STOP), EOFValidationError::unreachable_instructions);
+
+    add_test_case(
+        eof_bytecode(rjump(1) + OP_STOP + OP_STOP), EOFValidationError::unreachable_instructions);
+
+    // STOP reachable only via backwards jump - invalid
+    add_test_case(
+        eof_bytecode(rjump(1) + OP_STOP + rjump(-4)), EOFValidationError::unreachable_instructions);
```

Comment:
I think we wouldn't be able to distinguish really unreachable instructions from those reachable only via backwards jump (because we validate in a linear forwards pass)

So we could only make this error message more general, something like `instruction_not_reachabe_via_forward_control_flow` or even more general `invalid_instruction_order`. I don't have better ideas yet.

Reply:
Let's leave it for later.